api/v1alpha1/database_webhook.go

```diff
@@ -188,10 +195,43 @@ func instanceSpecValidatorForCreate(instance *Instance, allErrs field.ErrorList,
 		))
 	}
 
+	if err := additionalArgumentsValidationCheck(instance.Type, instance.AdditionalArguments); err != nil {
+		allErrs = append(allErrs, field.Invalid(instancePath.Child("additionalArguments"), instance.AdditionalArguments, err.Error()))
+	}
+
 	databaselog.Info("Exiting instanceSpecValidatorForCreate...")
 	return allErrs
 }
 
+/* Checks if configured additional arguments are valid or not and returns the corresponding additional arguments. If error is nil valid, else invalid */
+func additionalArgumentsValidationCheck(dbType string, specifiedAdditionalArguments map[string]string) error {
+	// Empty additionalArguments is always valid
+	if specifiedAdditionalArguments == nil {
+		return nil
+	}
+
+	allowedAdditionalArguments, err := util.GetAllowedAdditionalArgumentsForType(dbType)
+
+	// Invalid type returns error
+	if err != nil {
+		return err
+	}
+
+	// Checking if arguments are valid
+	invalidArgs := []string{}
+	for name, _ := range specifiedAdditionalArguments {
+		if _, isPresent := allowedAdditionalArguments[name]; !isPresent {
+			invalidArgs = append(invalidArgs, name)
+		}
+	}
+
+	if len(invalidArgs) == 0 {
+		return nil
+	} else {
+		return errors.New(fmt.Sprintf("Additional Arguments validation for database type: %s failed! The following args are invalid: %s. These are the allowed args: %s", dbType, errors.New(strings.Join(invalidArgs, ", ")), reflect.ValueOf(allowedAdditionalArguments).MapKeys()))
+	}
```

Comment:
Nit: We can extract the first arg (fmt.Sprintf..) into a variable for enhancing readability?

Reply:
Good catch, am using fmt.Errorf instead to help with readability.